### ISSUE TITLE
XS version of one() returns true when passed an empty list

### ISSUE THREAD

**Issue by randyl** [perl5-utils/List-MoreUtils#31]
The docs state:

> For an empty LIST, "one" returns false

However, I'm using perl 5.18, List::MoreUtils 0.428, and macOS 10.12, and I get the following:
```
$ perl -MList::MoreUtils -le 'print "defined" if List::MoreUtils::one { defined($_) } ();'
defined
$
```
The PP version works as expected:
```
$ perl -MList::MoreUtils::PP -le 'print "defined" if List::MoreUtils::PP::one { defined($_) } ();'
$
```

**Comment by petdance** [comment]
Even more concisely:

**Comment by petdance** [comment]
This also affects List::SomeUtils: https://github.com/houseabsolute/List-SomeUtils/issues/3

**Comment by petdance** [comment]
We mistakenly submitted this here instead of over RT, so I created one there: https://rt.cpan.org/Ticket/Display.html?id=123870

**Comment by rehsack** [comment]
Fixed in 0.428
